### DISCUSSION TITLE
Measurements from xDrip to nightscout

### DISCUSSION THREAD

**Discussion by DaNitroo** [NightscoutFoundation/xDrip#2701] [Q&A]
Hello, 
I would like to transfer the blood sugar I measured from my meter to nightscout. 

We use clarity, xdrip and nightcout and the accu check guides. 

The Xdrip data source is dexcom share and the nightscout data source is also dexcom share. The accu check guides transmit the measured values ​​to our cell phones and the values ​​also appear on all cell phones in xdrip. In Xdrip, the cloud upload settings are set so that the measured values ​​are to be transferred to dexcom share, the password and user name are stored correctly. But that doesn't work. 

The bloody measured values ​​that we see in xdrip are not transferred to dexcom or to nightscout :-(
Can someone help us there?

**Comment by Navid200** [comment]
What you describe means xDrip is neither the uploader to Nightscout nor the follower of Nightscout.
xDrip can upload to Nightscout in those two cases.

It may help if you explain why you need this and why you are using the setup that you have described and not an alternative.

**Reply by DaNitroo** [reply]
It`s correct. Both Xdrip and nightscout are my Dexcom Share followers. I didn't think about it any further and figured this would be the quickest way.

What do you mean what would be the quickest way?

I don't really care who the follower is. I would like to do whatever displays the data fastest and follow your tips. We use some M5Stack and they get their data from Nightscout, so it would be nice if Nightscout had up to date values ​​quickly and I thought this would be the quickest way...

**Reply by Navid200** [reply]
There are so many different options and so many different needs.  There is no one size fits all.
Can you tell us what CGM you use, what sensor type?

**Reply by DaNitroo** [reply]
Actually we use the Dexcom G7, the Dexcom Receiver and a Cubot King Kong Mini pro. The Cubot phone sent the data to dexcom share and xDrip and Nightscout are set as followers.

In June we become the Ypsopump and the G6 for 6 month because the g7 does not yet work with the ypsopump in a closed loop. 
The hospital says that we then have to use diasend or glooko and no more clarity.

I have seen the instructions for bringing here but have not dealt with it yet. I hope this can be incorporated into our setup...

Is it slower when I set nightscout as a Follower from xDrip?
Then we would see the measured blood sugar values ​​that we have in xDrip in nightscout, right?

**Reply by Navid200** [reply]
You can set xDrip to use Companion app mode.  Then, it will receive readings from your Dexcom app.
Then, you can set xDrip as your Nightscout uploader and the blood glucose measurements should also be uploaded to Nightscout.

When you switch to G6, you can use xDrip as the collector from G6.  And continue using xDrip as your uploader to Nightscout.

This explains how to set up xDrip to use the Companion app mode: https://navid200.github.io/xDrip/docs/Follow/CompanionApp.html

**Reply by DaNitroo** [reply]
OK I understand that.

I have now also set xDrip as my Nightscout uploader.

But what is the advantage of companion mode? I can also have Dexcom Share set up as a data supplier, right?

**Reply by DaNitroo** [reply]
The companion mode also works without internet i'm reading right now. I don't know if that's an advantage because the cell phone is always online anyway.

**Reply by Navid200** [reply]
If you need something that only you need because you are using a scheme that no one else does, your request will not get priority.
That is just for simplicity and reasonable practice in order to have a manageable app.
If you insist to use Dex share even though you don't need it, you can.  Not much I can do about that.

That's the only reason I have suggested that you use Companion mode, not because it doesn't need internet.
You still have not explained why you have chosen to use Dex share.  You don't have to.
All I'm saying is that there is a simpler solution for you, which should address what you asked for when you opened this discussion.

**Reply by DaNitroo** [reply]
i use dex share because i got it right away when i was diagnosed in the hospital and i thought it was the basis for everything else.

I assumed that the G7 would send the data to dexcom with the dex app and this would then be my basis for all other apps. And because it was so easy to set it up in xDrip with dex share, I left it that way.


If Companion Mode is a safe way then I will switch that. I'm primarily concerned with safety and reliability, since it's not me who has diabetes, but my 6-year-old son...


What setting do I have to make in the companion cell phone if I set the companion mode on my son's cell phone?


My and my wife's phones both have xDrip as well and are currently receiving data from Dex Share. It would be great if our cellphones could also transfer blood glucose readings from Xdrip to Nightscout. But that would probably be asking too much, right?

**Reply by DaNitroo** [reply]
My cell phone and my wife's could continue to receive the data from dex share.

Could I set all 3 mobile phones, i.e. my son’s, my wife’s and mine, as uploaders for Nightscout?

**Reply by Navid200** [reply]
xDrip is not a medical app.  
I have diabetes and I have chosen to use xDrip as the only app I use.  But, that's a decision I have made.

As a person who provides support to xDrip users, I cannot tell you, or anyone else, to use xDrip.
All I can do is to provide information that I have about it.
xDrip is maintained by a bunch of volunteers.  There are a small number of developers.  
So, we have to prioritize.

If you want to use official apps, there is nothing wrong with that.
We have had issues with xDrip uploading to Dex share before and the issues have been resolved.
But, I don't believe we have ever been able to upload blood glucose readings to Dex share.

Nightscout and xDrip are both open source.  xDrip sync is an xDrip feature that relies on Google.

Compare those to Dex share.  Dex share may not work if Dexcom changes something.  And, we cannot do anything about that.  We cannot contact Dexcom and ask them to change anything.

I hope you don't think that I want to convince you not to use Dex share.  You can use whatever you like.

When I mentioned the Companion mode app, it was as an option for replacing Dex share as the data source on your son's phone.  Then, you and your wife would use Nightscout as data source on your phones.
It was not meant to be something as an addition to what you have.

I'm not sure.  But, you can try setting up your phone as an uploader to Nightscout now.  It should then let you upload blood glucose readings to Nightscout.
But, I'm not sure what happens if you have multiple phones uploading the same data to Nightscout.  I don't know if there will be duplicates in Nightscout and if that may be a problem.

**Reply by DaNitroo** [reply]
I understand that and I'm totally enthusiastic about xDrip and think it's sensational that there are people who were able to develop something so great. I'm really grateful for that because xDrip is the best diabetes app in the world and without xDrip a lot of things in our lives would be more difficult!!

There is an upload function for blood glucose measurements. This can be found under Cloud Upload -> Dexcom Share> Upload Blood Glucose Results to Dexcom.

The nice thing about our current setting is that a measured blood sugar value that was transferred to one of our cell phones is also directly visible on all other cell phones. I don't know how that works, but I always thought that xDrip sends it to dex share and the other phones then get it back from dex share.

**Reply by Navid200** [reply]
I'm looking on the page you have described and the closest thing I can find on that page to what you have called is this:
![Screenshot_20230228-154436](https://user-images.githubusercontent.com/51497406/221975828-df752271-296b-47c6-b96d-9ae1b31182ae.png)

That is not related to blood glucose measurements performed on a meter.  BG stands for blood glucose.   True.  But, almost everywhere in xDrip, when you see BG, it is talking about readings that come from a sensor (a CGM), not a blood glucose meter.   

That's what I assume you have set in xDrip on your son's phone and that's how he is uploading to Dex share.
You cannot enable that on your phone or on your wife's phone in order to upload blood glucose meter readings from a meter.

**Reply by DaNitroo** [reply]
Yes, that's exactly the point I meant. oops Then I translated that wrong and misunderstood it! sorry


But there is one thing I don't understand in our configuration: Up until now, the cell phones were all dex sgare followers and each cell phone is paired with an Accu Check blood glucose meter. No matter which device we use to measure blood sugar, it transmits the Accu Check to a cell phone and the other cell phones also immediately have the measured blood sugar in xDrip. Excellent! But why is it like that? :-)

**Reply by Navid200** [reply]
I'm not sure.  
That's a question for you.
How have you set up the phones?

**Reply by DaNitroo** [reply]
Honestly, I have no idea :-)

But I'm glad it works!

I have set up all 3 phones as Dex share followers but if I click on -> System status and then on -> Followers, then there is always "xDrip sync group" and all of our 3 phones are under it, but I don't understand why and before xDrip knows them . I will test it...

**Reply by Navid200** [reply]
That means you are using xDrip sync.
It is under:

Settings &#8722;> xDrip+ Sync settings

**Reply by Navid200** [reply]
Can you look on your phone and see what it shows under:

Settings &#8722;> Hardware Data Source

**Reply by DaNitroo** [reply]
The point is actually activated and it says "master for followers".

Strange, how do you set that up? Aside from the dex share data, I never connected the phones. I do not understand that

**Reply by Navid200** [reply]
Please show what you have for hardware data source.

**Reply by DaNitroo** [reply]
![Screenshot_2023-02-28-22-36-41-261_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/122942359/221986397-b4805329-f6e4-407b-9ffd-b54ae6d2fc24.jpg)
![Screenshot_2023-02-28-22-36-17-326_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/122942359/221986403-90d09cc1-67db-4363-8a5b-3e0e48919609.jpg)
![Screenshot_2023-02-28-22-36-24-213_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/122942359/221986406-db4739d1-4023-451f-80ea-c571cbd3af90.jpg)
![Screenshot_2023-02-28-22-36-38-857_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/122942359/221986408-d6b1a4f2-3d1a-4419-95b1-d6e646b6ddda.jpg)

**Reply by DaNitroo** [reply]
Sorry for the big pictures - I´m a mechanican and no computer human ;-)

**Reply by Navid200** [reply]
Right,
All phones are set as masters for xDrip Sync.

It's very likely if you disable Dexcom share on your phone, it will still receive readings from your son's phone.

**Reply by DaNitroo** [reply]
strange, I didn't enable that, nor knowingly set it.

I'll test that tomorrow. Now it doesn't work because Junior is sleeping and that the phone is with hm ;-)

**Reply by DaNitroo** [reply]
I just tried to change the setup but unfortunately it doesn't make any sense. or I'm doing something wrong:

As soon as I switch my son's phone to companion mode in xDrip, it only gets the data 4-6 minutes later than with dexshare. Since he has a smartwatch that notifies him of low levels, that's quite a difference as fast as his blood sugar sometimes drops. Sometimes 5 minutes can make a difference.

![IMG-20230301-WA0045](https://user-images.githubusercontent.com/122942359/222238364-34f101dd-cd8c-4a85-911d-9b44dd70a0fa.jpg)
